### PR TITLE
Allow directory trees in dotfiles lists

### PR DESCRIPTION
**Pull request by yn**
Consider a directory tree layout in dotfiles layout like this:

```
dotfiles_files:
  - .gitconfig
  - .gitignore_global
  - .ssh/config
```

The current codebase supports this in a fairly straightforward manner, except it doesn't create parent directories like `.ssh` 

This PR fixes that by adding a play that creates parent directories.

### CODE REVIEW

**Comment by oxyc** [comment]
I already opened up a PR for this #12. That PR doesn't have `follow: yes` though. Is it necessary?

**Comment by yn** [comment]
It is not strictly speaking necessary, but it does allow you to have links
to subdirectories instead of subdirectories, if you pre-create them.

On Wed, May 31, 2017 at 6:55 AM, Oskar Schöldström <notifications@github.com

**Comment by oxyc** [comment]
@yn you should add `become: no` as well (geerlingguy often uses become:yes in the main playbook).

I can then close my PR and try and poke @geerlingguy to see if he can merge this :)

**Comment by yn** [comment]
@oxyc done

**Comment by cbracco** [comment]
Would also love to see this get merged! 😄 cc @geerlingguy ...anything in particular preventing this from happening? Happy to help if I can.

**Comment by tgallacher** [comment]
would love to see this merged

**Comment by jaulz** [comment]
@geerlingguy any chance to get this still merged? 😊

File: tasks/main.yml [hunk 1]
```diff
@@ -22,6 +22,14 @@
   when: "'@' not in item.1.stdout"
   with_indexed_items: "{{ existing_dotfile_info.results }}"
 
+- name: Ensure parent folders of link dotfiles exist.
+  file:
+    path: "{{ (dotfiles_home ~ '/' ~ item) | dirname }}"
+    state: directory
+    follow: yes
```
Comment (geerlingguy) [inline tasks/main.yml]:
```suggestion
    follow: true
```

File: tasks/main.yml [hunk 2]
```diff
@@ -22,6 +22,14 @@
...
+  become: no
```
Comment (geerlingguy) [inline tasks/main.yml]:
```suggestion
  become: false
```

**Review by geerlingguy — CHANGES_REQUESTED** [review]
A couple quick fixes for linting.

**Review by geerlingguy — APPROVED** [review]

**Comment by jaulz** [comment]
Wow, I didn't expect such a fast response. Thanks a lot for your work! ☺️

**Comment by geerlingguy** [comment]
Heh, honestly this PR just slipped through the cracks. Glad you brought it back up :)

**Comment by geerlingguy** [comment]
Hmm... after merging this in, I spotted an error in my mac-dev-playbook's CI run: https://github.com/geerlingguy/mac-dev-playbook/actions/runs/12173583322

```
TASK [Run .osx dotfiles.] ******************************************************
fatal: [localhost]: FAILED! => {"changed": false, "cmd": "/Users/runner/.osx --no-restart", "msg": "[Errno 13] Permission denied: b'/Users/runner/.osx'", "rc": 13, "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}
```

**Pull request merged**